File: rust/leetcode/src/merge_strings_alternately_1768.rs
pub fn merge_alternately(word1: String, word2: String) -> String {
    let word1 = word1.as_bytes();
    let word2 = word2.as_bytes();
    let mut i = (0,0);
    let mut rtn = String::from("");
    while i.0 < word1.len() || i.1 < word2.len() {
        if i.0 < word1.len() {
            rtn.push(word1[i.0] as char);
            i.0 += 1;
        } 
        
        if i.1 < word2.len() {
            rtn.push(word2[i.1] as char);
            i.1 += 1;
        }
    }
    return rtn;
}

File: java/LeetCode/src/leetcode/MergeStringsAlternately1768.java
package leetcode;

/**
 * MergeStringsAlternately1768
 */
public class MergeStringsAlternately1768 {

    public String mergeAlternately(String word1, String word2) {
        int i = 0, j = 0;
        char[] word1Array = word1.toCharArray();
        char[] word2Array = word2.toCharArray();
        StringBuilder sb = new StringBuilder();
        while (i < word1Array.length || j < word2Array.length) {
            if (i < word1Array.length) {
                sb.append(word1Array[i]);
            }

            if (j < word2Array.length) {
                sb.append(word2Array[j]);
            }

            i++;
            j++;
        }

        return sb.toString();
    }
}